codalab/model/bundle_model.py

```diff
@@ -157,8 +157,7 @@ def do_multirow_insert(connection, table, values):
         #   - Some dialects do not support multiple inserts in a single statement,
         #     which we deal with by using the DBAPI execute_many pattern.
         if values:
-            with connection.begin():
```

Comment:
This, to me, seems like a nested transaction (where `do_multi_row_insert` was called, the `connection` had always already had `begin()` called on it already). As such, I would actually think this would throw an error (see source code [here](https://github.com/sqlalchemy/sqlalchemy/blob/rel_2_0_19/lib/sqlalchemy/engine/base.py#L783-L857)), so I'm not sure why that wasn't happening.

Regardless, it would make sense this would introduce a race condition because this transaction becomes separate from the outer transaction that `do_multi_row_insert` is called in, and so other transactions can happen between this one and the 'outer' transaction.

Reply:
Great catch!